application/src/main/java/org/togetherjava/tjbot/config/FeatureBlacklist.java

```diff
@@ -32,4 +38,25 @@ public FeatureBlacklist(Set<T> featureIdentifierBlacklist) {
     public boolean isEnabled(T featureId) {
         return !featureIdentifierBlacklist.contains(featureId);
     }
+
+    /**
+     * Filters features stream to only having enabled features and logs any disabled features.
+     *
+     * @param features the feature stream to be filtered
+     * @param idExtractor function to get the class name of an object
+     * @return stream of features that are enabled
+     */
+    public <F> Stream<F> filterStream(Stream<F> features,
```

Comment:
```suggestion
    public <F> Stream<F> filterStream(Stream<? extends F> features,
```
This would allow more flexibility, similar to how you use `Function<? super F, ? extends T>`

Reply:
But this will make it return Stream<? extends T> but i want to only return Stream<T>. I could cast it but that would introduce unchecked casting warnings. I cant change the return type otherwise that would cause errors in other parts of the problem.
![filterStreamMethodError](https://github.com/Together-Java/TJ-Bot/assets/101219745/f899d824-1b8e-4c36-acdf-a35d17d8340e)

Reply:
@java-coding-prodigy sorry forgot to ping you.